int16 strategy
string task
int16 wp_index
